android/capacitor/src/main/java/com/getcapacitor/Plugin.java

```diff
@@ -46,10 +46,24 @@
     // Reference to the PluginHandle wrapper for this Plugin
     protected PluginHandle handle;
 
-    // A way for plugins to quickly save a call that they will
-    // need to reference between activity/permissions starts/requests
+    /**
+     * A way for plugins to quickly save a call that they will need to reference
+     * between activity/permissions starts/requests
+     *
+     * @deprecated use {@link #savedLastCallId} instead in conjunction with bridge methods
+     * {@link com.getcapacitor.Bridge#saveCall(PluginCall)},
+     * {@link com.getcapacitor.Bridge#getSavedCall(String)} and
+     * {@link com.getcapacitor.Bridge#releaseCall(PluginCall)}
+     */
+    @Deprecated
     protected PluginCall savedLastCall;
 
+    /**
+     * The call ID of a saved plugin call on the Bridge. This can be used to persist a call
+     * between activity lifecycle changes or permission requests.
+     */
+    protected String savedLastCallId;
```

Comment:
Do you think this still promotes an undesirable workflow where plugin devs are using a single variable to store multiple call IDs? I am almost of the opinion the plugin dev should be fully responsible for managing their call IDs. What do you think?

Reply:
We still need to store an ID for the last call for permission to work, but I'm currently refactoring this slightly to indicate its more tied to permissions, like `permissionsCallId`. Do you think that would make a good distinction that this is just used internally for permissions and it shouldn't be used by plugin author?

Im deprecating the `freeSavedCall`, `saveCall` and `getSavedCall` methods in addition to that, and making specific ones tied to permissions so the intent is clearer.

Thoughts?

Reply:
Yes, perfect!